packages/desktop-client/src/components/spreadsheet/format.ts

```diff
@@ -31,8 +41,23 @@ export default function format(value, type = 'string'): string {
         );
       }
 
-      return integerToCurrency(value);
+      return integerToCurrency(value, formatter);
     default:
       throw new Error('Unknown format type: ' + type);
   }
 }
+
+export function useFormat() {
+  const numberFormat = useSelector(state =>
+    getNumberFormat({
+      format: state.prefs.local.numberFormat,
+      hideFraction: state.prefs.local.hideFraction,
+    }),
+  );
```

Comment:
Since `getNumberFormat` returns an object literal, this will return a different value on every render and cause excessive rerendering. I would recommend fetching each of the state values separately, then using `useMemo` to construct the number format.

Reply:
Good shout. Added `reselect` that will take care of this. We should port all other selectors to it too.